zkvm/src/vm.rs

```diff
@@ -307,11 +307,7 @@ where
 
     fn r#const(&mut self) -> Result<(), VMError> {
         let data = self.pop_item()?.to_data()?.to_bytes();
-        let mut slice = Subslice::new(&data);
-        let scalar = slice.read_scalar()?;
-        if slice.len() != 0 {
-            return Err(VMError::FormatError);
-        }
+        let scalar = SliceReader::parse(&data, |mut r| r.read_scalar())?;
```

Comment:
ditto, if reader is passed by &mut

```suggestion
        let scalar = SliceReader::parse(&data, |r| r.read_scalar())?;
```